lib/src/text-input/TextInput.tsx

```diff
@@ -94,31 +70,28 @@ const DxcTextInput = React.forwardRef<RefType, TextInputPropsType>(
       tabIndex = 0,
     },
     ref
-  ) => {
-    const [inputId] = useState(`input-${uuidv4()}`);
-    const [innerValue, setInnerValue] = useState(defaultValue);
+  ): JSX.Element => {
+    const id = useState(uuidv4());
+    const inputId = `input-${id}`;
+    const autosuggestId = `suggestions-${id}`;
+    const errorId = `error-${id}`;
 
+    const [innerValue, setInnerValue] = useState(defaultValue);
     const [isOpen, changeIsOpen] = useState(false);
     const [isSearching, changeIsSearching] = useState(false);
     const [isAutosuggestError, changeIsAutosuggestError] = useState(false);
     const [filteredSuggestions, changeFilteredSuggestions] = useState([]);
-    const [visualFocusedSuggIndex, changeVisualFocusedSuggIndex] = useState(-1);
+    const [visualFocusIndex, changeVisualFocusedSuggIndex] = useState(-1);
```

Comment:
Rename `changeVisualFocusedSuggIndex` to `changeVisualFocusIndex`.

Reply:
Agreed on this one!